### PR TITLE
NEW: Disable schema introspection in non-dev

### PR DESCRIPTION
**Pull request by unclecheese**
Disables schema introspection outside of dev. Tests to follow.

### CODE REVIEW

**Comment by madmatt** [comment]
Thanks @unclecheese, from my very uneducated eye this looks good to me! :)

File: README.md [hunk 1]
```diff
@@ -2502,13 +2502,36 @@ a `types.graphql` file will be written to your `assets/` directory on `flush`.
...
+the [graphql IDE](https://github.com/silverstripe/graphql-devtools).
+
+If you would like to enable schema introspection outside of your dev environment, you can set the `allowIntrospection`
+property on the `Manager` instance.
+
+```html
+SilverStripe\Core\Injector\Injector:
+  SilverStripe\GraphQL\Manager.default:
+    properties:
+      AllowIntrospection: true
```
Comment (michalkleiner) [inline README.md:2530]:
```suggestion
      allowIntrospection: true
```

**Comment by madmatt** [comment]
I merged your suggested change @michalkleiner, this keeps it consistent with the rest. I'm keen to get this merged in, however I just noticed there's a conflict in the README.md file now and realise that there's already a way to remove introspection, it's just done in code and for all environments (though you can use code to amend that).

See https://github.com/silverstripe/silverstripe-graphql#schema-introspection.

Does that invalidate the need for this PR perhaps?

**Comment by madmatt** [comment]
Actually on second thought I think that existing feature is designed to stop the _dynamic_ endpoint that shows the introspection dynamically. The point is to make it static as a file in `assets/` instead, so this is still requierd.

**Comment by madmatt** [comment]
@unclecheese Do you mind if I get this conflict fixed and remove the draft status of this PR, building on top of your work? I've got a few different clients interested in seeing something like this merged in.

Thanks!

**Comment by unclecheese** [comment]
Yeah, go for it!

**Comment by GuySartorelli** [comment]
@madmatt are you still keen to fix the conflict on this PR?

**Comment by GuySartorelli** [comment]
Given that this is still in draft, has conflicts, and is for v3 which we are no longer actively developing, I'm going to close this.
If someone is really passionate about getting this merged in, feel free to open a new PR with no merge conflicts.

**Pull request closed without merging**